optimum/neuron/utils/cache_utils.py

```diff
@@ -266,9 +267,13 @@ def get_num_neuron_cores() -> int:
         os.environ["PATH"] = path
     proc = subprocess.Popen(["neuron-ls", "-j"], stdout=subprocess.PIPE)
     stdout, _ = proc.communicate()
-    stdout = stdout.decode("utf-8")
-    json_stdout = json.loads(stdout)
-    return sum(neuron_device_info["nc_count"] for neuron_device_info in json_stdout)
+    if proc.returncode != 0:
+        num_cores = 0
+    else:
+        stdout = stdout.decode("utf-8")
+        json_stdout = json.loads(stdout)
+        num_cores = sum(neuron_device_info["nc_count"] for neuron_device_info in json_stdout)
+    return num_cores
```

Comment:
I think you can also simply count the number of pseudo-devices under /dev, like I did here: https://github.com/huggingface/optimum-neuron/blob/a00d42513765adb4d78eaeabf36fc4b89d4a92eb/optimum/neuron/modeling_decoder.py#L211

Reply:
Nice solution!
Better than what I had done, thanks!